Presentation.Web/app/services/gridStateService.ts

```diff
@@ -325,16 +370,174 @@
                 $window.localStorage.removeItem(storageKey);
+                $window.localStorage.removeItem(locallyChangedKey);
+                $window.localStorage.removeItem(nonExistingOrganizationalConfigurationColumnsKey);
+                removeOrgConfig();
             }
 
             function removeProfile(): void {
                 $window.localStorage.removeItem(profileStorageKey);
             }
+
+            function removeOrgConfig(): void {
+                $window.localStorage.removeItem(organizationalConfigurationColumnsKey);
+                $window.localStorage.removeItem(organizationalConfigurationVersionKey);
+            }
+
+            function doesGridDivergeFromOrganizationalConfiguration(overviewType: Models.Generic.OverviewType, grid: Kitos.IKendoGrid<any>): boolean {
+                if (overviewType === null || overviewType === undefined) {
+                    return false; // No defaults defined for unknown overview type
+                }
+
+                if ($window.localStorage.getItem(organizationalConfigurationColumnsKey) === null) {
+                    return false;
+                }
+
+                if (grid === undefined || grid === null) { // The grid is not always initialized the first time this code is called.
+                    return false;
+                }
+
+                if (gridLoading) { // If grid is still loading we can't correctly calculate if it is deviating
+                    return false;
+                }
+
+                var options = grid.getOptions();
+                return !isOrgConfigServerVersionEqualToLocalGrid(options);
+            }
+
+            function isOrgConfigServerVersionEqualToLocalGrid(options: Kitos.IKendoGridOptions<any>) {
+                const localVersion = computeGridVersion(options);
+                const organizationalConfigurationVersion = $window.localStorage.getItem(organizationalConfigurationVersionKey);
+                if (organizationalConfigurationVersion !== null && localVersion !== organizationalConfigurationVersion) {
+                    return false;
+                }
+                return true;
+            }
+
+            function computeGridVersion(options: Kitos.IKendoGridOptions<any>) {
+                var gridColumnsPersistIds = options.columns.filter(x => !x.hidden).map(x => x.persistId);
+
+                // Add items from the server which doesn't exist in kendogrid to be able to calculate the same version as the backend.
+                var nonExistingColumnsItem = $window.localStorage.getItem(nonExistingOrganizationalConfigurationColumnsKey);
+                if (nonExistingColumnsItem) {
+                    var nonExistingColumns = <string[]>JSONfn.parse(nonExistingColumnsItem, true);
+                    gridColumnsPersistIds.pushArray(nonExistingColumns);
+                }
+
+                return sha256(gridColumnsPersistIds.sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase())).join(""));
```

Comment:
sikrer du at du comparer på samme måde på serveren?.. det her er meget specifikt og du bruger

Reply:
Den er meget specifik i frontend efter jeg oplevede at frontend sorterede anderledes end backend. Backend var ikke case sensitive mens frontend var det. Jeg retter lige i backend så den helt sikkert også kører toLowerCase()

Reply:
super

Reply:
spørgsmålet er om du skal tvinge den på en bestemt locale både i frontend og backend for at sikre at sorteringen altid sker efter de samme regler?

Du kan jo også smide locale på i backend. Lige nu er det "held" der afgør om den er korrekt (server locale = client locale)

Reply:
"localeCompare" bruges til at omdanne string til number som er det eneste der fungerer i sort. Uden andre argumenter sammenligner den bogstaverne i rækkefølge.

Reply:
alright jamen så ska ldet bare være tolower både backend og frontend :)